internal/resource/role/model.go

```diff
@@ -0,0 +1,67 @@
+package role
+
+import (
+	"context"
+
+	sgsdkgo "github.com/StackGuardian/sg-sdk-go"
+	flatteners "github.com/StackGuardian/terraform-provider-stackguardian/internal/flattners"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type RoleAllowedPermissionsPathsModel []string
+
+type RoleAllowedPermissionsModel struct {
+	Name  types.String `tfsdk:"name"`
+	Paths types.Map    `tfsdk:"paths"`
+}
+
+type RoleResourceModel struct {
+	Organization       types.String `tfsdk:"organization"`
+	ResourceName       types.String `tfsdk:"resource_name"`
+	Description        types.String `tfsdk:"description"`
+	AllowedPermissions types.Map    `tfsdk:"allowed_permissions"`
+	Tags               types.List   `tfsdk:"tags"`
+}
+
+func (m *RoleResourceModel) ToAPIModel(ctx context.Context) (*sgsdkgo.Role, diag.Diagnostics) {
+	var allowedPermissionsModelValue *RoleAllowedPermissionsModel
+	var allowedPermissions = m.AllowedPermissions.ElementsAs(context.Background(), &allowedPermissionsModelValue, false)
+	if allowedPermissions.HasError() {
+		return nil, allowedPermissions
+	}
+	var allowedPermissionsPathsModelValue *RoleAllowedPermissionsPathsModel
+	var allowedPermissionsPaths = allowedPermissionsModelValue.Paths.ElementsAs(context.Background(), &allowedPermissionsPathsModelValue, false)
+	if allowedPermissionsPaths.HasError() {
+		return nil, allowedPermissionsPaths
+	}
+
+	apiModel := sgsdkgo.Role{
+		ResourceName:       m.ResourceName.ValueStringPointer(),
+		Description:        m.Description.ValueStringPointer(),
+		Tags:               m.Tags,
+		AllowedPermissions: allowedPermissions,
```

Comment:
allowedPermissionsPaths are not assigned to the struct allowedPermissions.

Similar to tags you'll need to iterate over the Map values and convert them to go types.